ui/src/components/pipeline/Pipeline.tsx

```diff
@@ -66,9 +71,35 @@ export function Pipeline() {
     }
   }, [pipeline]);
 
+  // This useEffect is used to obtain all the rates for a given vertex in a pipeline.
+  useEffect(() => {
+    const vertexToRateMap = new Map();
+
+    if (pipeline?.spec?.vertices) {
+      Promise.all(
```

Comment:
Not blocking, but may be beneficial to update server side to make one batched network call instead of many calls.

Reply:
That's a good point, I think I can tackle this in the next PR